include/boost/geometry/views/detail/normalized_view.hpp

```diff
@@ -36,72 +36,99 @@ namespace boost { namespace geometry {
-                order_as_direction
-                    <
-                        geometry::point_order<Geometry>::value
-                    >::value
-            >::type reversible_type;
-
-    typedef std::conditional_t
-        <
-            is_const,
-            reversible_type const,
-            reversible_type
-        > reversible;
-
-    typedef typename
-        closeable_view
-            <
-                reversible,
-                geometry::closure<Geometry>::value
-            >::type closeable_type;
-
-    typedef std::conditional_t
+private:
+    view m_view;
+};
+
+
+template
+<
+    typename Geometry,
+    closure_selector Closure = geometry::closure<Geometry>::value,
+    iterate_direction Direction = geometry::order_as_direction
         <
-            is_const,
-            closeable_type const,
-            closeable_type
-        > closeable;
+            geometry::point_order<Geometry>::value
+        >::value
+>
+struct close_reverse_view
+{
+    // These utilities are inconsistent since Closure defines the property of
+    // the Geometry while Direction defines what to do with the Geometry.
+    using close_view = detail::close_view<Geometry const, Closure>;
+    using view = detail::reverse_view<close_view const, Direction>;
     
-    explicit inline normalized_view(range & r)
-        : m_reversible(r)
-        , m_closeable(m_reversible)
+    explicit inline close_reverse_view(Geometry const& g)
+        : m_view(close_view(g))
     {}
 
-    typedef typename boost::range_iterator<closeable>::type iterator;
-    typedef typename boost::range_const_iterator<closeable>::type const_iterator;
+    using iterator = typename boost::range_iterator<view const>::type;
+    using const_iterator = typename boost::range_iterator<view const>::type;
 
-    inline const_iterator begin() const { return boost::begin(m_closeable); }
-    inline const_iterator end() const { return boost::end(m_closeable); }
-
-    inline iterator begin() { return boost::begin(m_closeable); }
-    inline iterator end() { return boost::end(m_closeable); }
+    inline const_iterator begin() const { return boost::begin(m_view); }
+    inline const_iterator end() const { return boost::end(m_view); }
 
 private:
-    reversible_type m_reversible;
-    closeable_type m_closeable;
+    view m_view;
 };
 
+
+// Other possible names:
+//   clockwise_closed_view
+template
+<
+    typename Range,
+    order_selector Order = geometry::point_order<Range>::value,
+    closure_selector Closure = geometry::closure<Range>::value
+>
+using clockwise_close_view = reverse_close_view
+    <
+        Range,
+        geometry::order_as_direction<Order>::value,
+        Closure
+    >;
+
+
+// Other possible names:
+//   closed_clockwise_view
+template
+<
+    typename Range,
+    closure_selector Closure = geometry::closure<Range>::value,
+    order_selector Order = geometry::point_order<Range>::value    
+>
+using close_clockwise_view = close_reverse_view
```

Comment:
Same here, why do we need it, and why do we need two variants in different orders? First reversing and then closing should have the same effect as closing and then reversing... or are there exceptions here?

Reply:
I described this above, but to reiterate.

> Same here, why do we need it, and why do we need two variants in different orders?

Because in some algorithms `Reverse` parameter is passed and in other algorithms `Order` is taken from Geometry, so to avoid translating them to `Direction` in each algorithm this is done here once.

> First reversing and then closing should have the same effect as closing and then reversing... or are there exceptions here?

The difference is that when we first reverse and then close the first point is not the same as in the original range. Geometrically they are the same, but the results of some algorithms will be different. Currently we have algorithms performing these operations in two different orders and this PR is not intended to change that. It's because I didn't want to change the results of e.g. set operations or buffer. This might also interfere with your rescaling changes. I think that we should change that in the future though but this requires testing.

Reply:
One more thing, regarding:

> It's because I didn't want to change the results of e.g. set operations or buffer for opened rings.

The exception is `convert()` which I modified in this PR to preserve the first point for opened rings which makes the result strictly equal to the one for closed rings.

Reply:
On second thought. Correct me if I'm wrong. AFAIR we mainly test algorithms for clockwise, closed geometries anyway. So I think it would be quite safe to change the order of closing and reversing even now. If anything this would make the results more consistent to what we test against. So I think I could change it now and remove the redundant views (`reverse_close_view` and `clockwise_close_view`) in this PR. What do you think?

Reply:
Was this question still relevant?
I'm fine with it.

Reply:
Not relevant, this was already done.